test_api.py

```diff
@@ -0,0 +1,27 @@
+import json
+from src.server import create_app
+from src.DataFetcher import fetchDataForTickerSymbol
+
+
+def mock_fetch_data():
+    return {}
+
+def test_import_app():
+    app = create_app(fetchDataForTickerSymbol)
+
+    with app.test_client() as test_client:
```

Comment:
I'm not too familiar with writing tests for Flask (i'm not a python dev by trade, heh).

This is for hermetic tests, where we provide mock responses. Right?

(I also imagine a world where we write some tests that actually fetch data, which may be flaky as they depend on network. but might work as a good way to detect 'is this endpoint still live') I think ideally we want both:

1. Does our parsing work given canned data (hermetic, quick, reliable)

2. Does fetching the data over the network still reply with a payload that has valid data and not 404 or a different data format.

Reply:
> This is for hermetic tests, where we provide mock responses. Right?

Yes.

> I think ideally we want both

I agree. I may even try to include these two kinds of tests here in this PR. I have an idea.

Reply:
I think it's alright to do it in a separate PR just to keep the scope small.

This one looks good so let's land it.